### PR TITLE
Update Carbon Kernel version

### LINKED ISSUE #1253: Certificate issue while trying to log in to the MI Monitoring Dashboard
**Description:**
When trying to login to MI Monitoring Dashboard with a self-signed certificate the following error occurs,

image

We could observe the following logs in the dashboard console, 

```
ERROR {org.wso2.transport.http.netty.contractimpl.listener.WebSocketServerHandshakeHandler} - Error during WebSocket server handshake io.netty.handler.codec.DecoderException: javax.net.ssl.SSLException: Received fatal alert: bad_certificate
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:459)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:265)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1359)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:935)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:141)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:645)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:580)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:497)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:459)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:886)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
```

To overcome this issue we need to add the certificate of the micro integrator instance to the browser as a trusted source. We may direct the browser to `https://localhost:9164/management` and add the site as trusted (given MI hostname is localhost). Then we should be able to login to the MI dashboard with the host `localhost`. 
If we add the micro integrator instance to the browser as a trusted source by directing the browser to `https://{some_ip}:9164/management` then, the MI dashboard should be logged in with host `some_ip`.

### PR DESCRIPTION
**Pull request by sajinieKavindya**
## Purpose
- Update carbon kernel version to include the fix done for wso2/carbon-kernel#2327
- Update and refactor test modules' poms.

Fixes #1253

### CODE REVIEW

File: integration/cli-tests/pom.xml [hunk 1]
```diff
@@ -120,6 +120,54 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <artifactId>maven-resources-plugin</artifactId>
+                <version>2.6</version>
```
Comment (Nirothipan) [inline integration/cli-tests/pom.xml:125]: Shall we not hard code the version here and inherit from parent pom

File: integration/dataservice-hosting-tests/tests-integration/tests/pom.xml [hunk 2]
```diff
@@ -123,6 +123,26 @@
...
+                        <configuration>
+                            <artifactItems>
+                                <artifactItem>
+                                    <groupId>org.wso2.ei</groupId>
+                                    <artifactId>wso2mi</artifactId>
+                                    <version>${project.version}</version>
+                                    <type>zip</type>
+                                    <overWrite>true</overWrite>
+                                    <outputDirectory>${basedir}/target/tobeCopied/</outputDirectory>
+                                    <includes>**/*.jks,**/*.mar,**/synapse_sample*.xml</includes>
```
Comment (Nirothipan) [inline integration/dataservice-hosting-tests/tests-integration/tests/pom.xml:141]: Only jks is enough ?

File: integration/mediation-tests/tests-transport/pom.xml [hunk 3]
```diff
@@ -88,6 +88,26 @@
...
+                        <configuration>
+                            <artifactItems>
+                                <artifactItem>
+                                    <groupId>org.wso2.ei</groupId>
+                                    <artifactId>wso2mi</artifactId>
+                                    <version>${project.version}</version>
+                                    <type>zip</type>
+                                    <overWrite>true</overWrite>
+                                    <outputDirectory>${basedir}/target/tobeCopied/</outputDirectory>
+                                    <includes>**/*.jks,**/*.mar,**/synapse_sample*.xml</includes>
```
Comment (Nirothipan) [inline integration/mediation-tests/tests-transport/pom.xml:106]: Same here .

File: integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/ssl/test/DynamicSSLProfilesPTTListenerTestCase.java [hunk 4]
```diff
@@ -52,6 +52,10 @@
     private String configLocation =
             getESBResourceLocation() + File.separator + "dynamicsslprofiles" + File.separator + "pttlistener"
                     + File.separator;
+    private final String CERT_CONFIG_LOCATION =
+            FrameworkPathUtil.getSystemResourceLocation() + File.separator + "keystores" + File.separator + "products"
```
Comment (Nirothipan) [inline integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/ssl/test/DynamicSSLProfilesPTTListenerTestCase.java:56]: FrameworkPathUtil.getSystemResourceLocation()  give the path with File.Separator in the end , extra separator might some time cause issue in windows.

File: integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/ssl/test/DynamicSSLProfilesPTTListenerTestCase.java [hunk 5]
```diff
@@ -16,23 +16,23 @@
 
 package org.wso2.carbon.esb.ssl.test;
 
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.management.MalformedObjectNameException;
```
Comment (Nirothipan) [inline integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/ssl/test/DynamicSSLProfilesPTTListenerTestCase.java:23]: Usually we have the java imports at the end .

File: integration/cli-tests/pom.xml (hunk 1 above)
Comment (sajinieKavindya) [reply integration/cli-tests/pom.xml]: Done.

File: integration/dataservice-hosting-tests/tests-integration/tests/pom.xml (hunk 2 above)
Comment (sajinieKavindya) [reply integration/dataservice-hosting-tests/tests-integration/tests/pom.xml]: Done

File: integration/mediation-tests/tests-transport/pom.xml (hunk 3 above)
Comment (sajinieKavindya) [reply integration/mediation-tests/tests-transport/pom.xml]: Done

File: integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/ssl/test/DynamicSSLProfilesPTTListenerTestCase.java (hunk 4 above)
Comment (sajinieKavindya) [reply integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/ssl/test/DynamicSSLProfilesPTTListenerTestCase.java]: Done

File: integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/ssl/test/DynamicSSLProfilesPTTListenerTestCase.java (hunk 5 above)
Comment (sajinieKavindya) [reply integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/ssl/test/DynamicSSLProfilesPTTListenerTestCase.java]: Done

**Pull request merged**